### DISCUSSION TITLE
Turn on / off screen in navigation

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#61] [Q&A] [answered]
For LCD the Cruiser screen saver (oled) option does not save energy.
For LCD the on_off mode by the "accessibility services" is necessary.

FAQ and info by the [WaveUp](https://gitlab.com/juanitobananas/wave-up/-/blob/3c296b4e6d40d52b2c2a16642978f43c6249d756/FAQ.md#faq) proximity app.  
**- Display OFF/ON is possible without device admin rights this since A9.**

Proximity sensor.  
_The extra current drainaige to activate the proximity sensor is LOW_ 
_Consumes way more LESS than the power consumed by the LCD display_

- Single wave over the sensor: SET DISPLAY ON. (constant)
- Cover the sensor +1.5 sec:  SET AUTOMATIC. (auto on/off)

See how it works:  https://youtube.com/shorts/6Iyez-hVeGw

**Comment by devemux86** [comment]
Thanks for the interesting suggestion!

It is already in my implementation list. :slightly_smiling_face:

**Reply by devemux86** [reply]
> Deleted in github ?

I read the description better, so my question was no longer needed. :slightly_smiling_face: 

Thanks for the extra details.
Indeed, Android is strict with wake / unlock permissions, they are usually allowed for "alarm" apps.

**Comment by JimmyS83** [comment]
This is nice idea!  As owner of older (slower and no OLED display) I can confirm, that LCD is vast majority part of consumed energy when navigating.

Could it be, that power features has also option to (optionally, or insted of dimming) control turning screen on and off (off could be handled by Android screen timeout, but ON need to be triggered from app)? 

My experience is, that
_Dim backlight to minimum is only a little bit less power efficiënt than display off_
is not 100% true in all cases. Its not valid for my phone(s), at night I am using 0% brightness, and still, have display turned off is MAJOR difference for battery, comapring to display ON.

_Avoid annoying PIN unlock requests._
My navigation dedicated phone has no PIN enabled just to be quickly operable with gloves through buttons

_+1,5 sec_
My old phone takes like 2-5 sec. just to turn screen ON and show something, and then you would like to have actually have some time to orient in map before turn (in case of city multiways) I would vote for slider more than constant here :-D 


To be not understood, I support 100% this dim idea, and for OLED it could work just fine with PIN enabled, but when we would talk about power saving, I think it worth think more complex, and give a shot also for older mobiles - especially, when older mobiles are more suitable for use on motorcycle handlebar than new expensive mobiles with optic stabilization :)  (in case, that app actually **could** send wake screen signal of course.)

**Reply by JimmyS83** [reply]
As I wrote, I really thought that GPS is also drainer, but in my case, LCD off saves the most..

I was really suprprised, how the time was prolonged, when I test it by just bluetooth and voice navigation without display.. Currently, sometimes, when I dont want to constantly recharge, I just turn on/off by manually via power button press during trip with my motogloves, and its really difference when comes to battery..

For me, I dont think my phone has any proximity sensor, but I maybe explore those options as well, would be nice to use wave instead pushing button, but not sure if I will be able to use that. Anyway, GPS must be on neverthless, so I dont its relevant when comes to power saving discussion in Navi software..

0% brightness was meant as 0% of brightness slider in android. Lowest possible brightness you can get.

**Reply by devemux86** [reply]
Note that keeping the proximity sensor on is not good for the battery.
(unless the device has native such wake-up functions)

Also, admin rights may have worked in old Android, now this is restricted and not allowed to publish on Google Play.

**Reply by 0709wiwiwi** [reply]
> Note that leaving the proximity sensor on is bad for the battery.

_It's more of an excuse not to have to offer this feature ?._
The extra drainage is **minimal**.  The sensor halts when the app closes. 
See also the updated information at the idea proposal start. 
_From A9 and on device **admin rights** are no longer necessary_

**Reply by JimmyS83** [reply]
Turns out i have proximity sensor :) I test wave up application (https://gitlab.com/juanitobananas/wave-up) and it works great, thanks for tip! If Cruiser would have option to send the same on off signals around turn direction, it would be superb :)  And agree, as i dont know, how much drainage proximity monitoring takes, it comes to my mind. that wouldnt be the case of cruiser though..

**Reply by Ciutnik** [reply]
> I did so add some older test pictures

Cool! Can you reveal how you did it? Some application or traditional: meter, cables and soldering iron? 🙂

**Reply by 0709wiwiwi** [reply]
Old App: Battery Charging Current Monitor v 1.03 . = (Removed in current Google play) 
New app: Battery Charging monitor.  https://play.google.com/store/apps/details?id=com.tofabd.batteryanalyzer

**Reply by devemux86** [reply]
Can also search in websites like APKPure, APKMirror, etc.

**Comment by JimmyS83** [comment]
Hello, is there a chance to have some sort of On/Off/Dim screen, based on routing/navigating activies, or is that rule out due to android permissions? So far I sucessfully using waving around handlebars to save battery (great tip), but have it done automatic from the Cruiser idea is still just as tempting as it was when I first read it :)

**Reply by devemux86** [reply]
@JimmyS83 this is a complicated feature and indeed there are Android restrictions.

So it needs some time to be implemented. :slightly_smiling_face: 

---

Cruiser already has a screen saver (in navigation settings), it can be useful especially with OLED screens.

**Reply by JimmyS83** [reply]
> Cruiser already has a screen saver (in navigation settings), it can be useful especially with OLED screens.

Overlooked this one during waiting.. Does this work with Always screen ON option? Because if I dont enable Always screen ON, my phone went locked in XX seconds, and needs to be unlocked via biometrics after.  When I keep Always ON, and set screen saver after 5sec, when running simulation on higway without near instructions, there seems to be no change on display after 5sec.

BTW., in some of recent version of Waze, its first navigation I went along on Android, which is able to "override" lock screen: New option "Navigate with locked screen" allows turn screen OFF and then ON, without need to unlock biometrically to actually see (with map and all) navigation. No rooted phone, or fishy things, and it seems to work.. That would be great addition to potentional "wake/un-dim screen before action notification + lock screen/or dim black when nothing happening XXsec" feature, which I desperate try to find :)

Btw., I tested like 8 navigation apps, and in regards of "staying alive and working on locked phone", Cruiser does the **best job** of all of them. Dunno why others are so unreliable (despite battery optimiser tuning, locking apps to be alive, no internet restrictions, only few could survive and usually they need to be in navigation mode), but Cruiser is rock stable and solid!

**Reply by devemux86** [reply]
@JimmyS83 

"Screen saver" can work together with "Keep screen on".

"Screen saver" works in real navigation (not simulation) and follow location modes.

**Reply by JimmyS83** [reply]
thanks! will test it! With Fullscreen mode ON it could save a lot of battery for OLEDs, is there some other conditions to wake the screen apart touching it? I mean, if there is need for setting voice announcement or something for navigation mode..

**Reply by WolfgangZ42** [reply]
>  is there some other conditions to wake the screen apart touching it? 

At each announcement the screen is automatically turn on. I use it on my bicycle tours (no charging available) and it works like a charm!

**Reply by JimmyS83** [reply]
> At each announcement the screen is automatically turn on. I use it on my bicycle tours (no charging available) and it works like a charm!

Thanks! Thats very nice! So the actual Voice announcement needs to be enabled, right? It doesnt turn on just base on "yellow dot" in map (dunno how to call it, direction point or something like that, I found thats the point where voice announcement is made)?

I would adapt my some of my "visual to handlebars" profiles and run for some tests in summer for sure.. Would be interesting to see, how it would comapare to my usual "wave up - lock/unlock" handlebar profile in terms of battery when using Amoled :)

**Reply by devemux86** [reply]
@JimmyS83 

> "yellow dot" in map (dunno how to call it, direction point or something like that, I found thats the point where voice announcement is made)?

The yellow dots are the turn instructions that are announced in navigation.

> So the actual Voice announcement needs to be enabled, right?

There are voice announcements for turn instructions and nearby POI + favorites.
You can customize freely what voice announcements to hear and their distances.

**Reply by JimmyS83** [reply]
Yes, I have quite a customized announcement setup, for mix both of favourites and pois :) They worked really well, my question was rather to "what wakes up" the screen from screen saver mode (which I cant test right now in real navigation).. If there needs to be voice announcement ON, or would it trigger by "yellow dots" (that would be working for non-voice anouncement profiles as well). I know Wolfgang wrote about announcements, so I assume second option (which is also more versatile), but asked anyway :)

To be honest, I cant wait to test screensaver.. I was quite happy with one phone in pocket, safe from rain and serving just Voice instructions, and second, optional handlebar "visual navigation" without voice announcement, on/off controlled via proximity sensor in dense areas, where visual is "safer". Works fine with gloves, saving battery A LOT, but I need to have phone lock completelly disabled for handlebar phone, as proximity sensor app couldnt unlock non-rooted phone :(

**Reply by devemux86** [reply]
> If there needs to be voice announcement ON, or would it trigger by "yellow dots"

The yellow dots are just the locations of turn instructions on the map.

Voice announcements are independent from the map appearance.
They can be turned on / off and you can customize their distances and at how many distances you hear them.

Screen saver activates the screen when something requires your attention.
It could be a turn instruction in 1.5 km or 300 m or a nearby POI / favorite.

So it depends on what you have enabled in voice guidance settings.

**Reply by JimmyS83** [reply]
I can also confirm it works very well, just tested on the car. Not enough to make any power comparation with screen turned off, but this is really handy thing which I really like! :)

**Comment by JimmyS83** [comment]
Ok, I understand it. I just wanted to know, if there is still hope for the future, nothing more :)

**Comment by 0709wiwiwi** [comment]
[WaveUp](https://play.google.com/store/apps/details?id=com.jarsilio.android.waveup)
[Display on_off ](https://youtube.com/shorts/2RfEYTl8VCU)

Some apps do not offer LCD power savings at all.
They show a black screen without backlight off.
Check by viewing the screen in total darkness.
With backlight on some light will seep through.

**Reply by JimmyS83** [reply]
This seems to be very similiar what I described in recent version of Waze.. In Waze case, phone is still locked, but when you trigger unlock (wave up or button), instead of unlocking screen waze app is showed.. (Android 13, not sure about compatibility with other-older versions).  But insteresting option for sure.

**Comment by devemux86** [comment]
Next version will be able to display Cruiser on the lock screen without having to unlock the device,
when Cruiser runs and we press the power button to turn on the screen.

---

Also we will have a new setting "**Turn on screen**" that will work in navigation.

- Disable the "**Keep screen on**" option, so the screen can be switched off by the system screen timeout

- Enable the "**Turn screen on**" option, so the screen can be switched on automatically in navigation

- It uses the system screen timeout to turn off completely the screen
(customized in Android display settings) 

- Navigation can automatically turn on the screen in all navigation events

- _Depending on the Android version and device you may need to use at most the Swipe lock option_
_(not pattern, pin, password, etc.)_

- It can also be tested in navigation simulation

**Reply by JimmyS83** [reply]
Thanks! Looking forward to it. I am thinking that my security PIN/print would prevent that, but it could help in some use cases when that kind of security is not needed (navi dedicated devices), or android version will be friendly. To be hones, screensaver function saves me totally, well my battery :)

**Reply by devemux86** [reply]
- Cruiser on lock screen and automatic turn on screen are available in https://github.com/devemux86/cruiser/discussions/829#discussioncomment-13821971

Please read the instructions [above](https://github.com/devemux86/cruiser/discussions/61#discussioncomment-13816480) and let me know how it works on your devices.

**Reply by 0709wiwiwi** [reply]
To save maximum energy (cycling) I do not set screen wake up at turns. 
Cruiser Keep screen on can so still be activated.

Successful tests on both A10 & A13.
A. By power button. (Button poorly accessible in the bike holder)
B. By the proximity Wave Up app with the next set up.
- Cover the sensor +1,5 s > sleep.
- Wave once over the sensor > wake up.
App that triggers to sleep can usually trigger a wake up.  
Someone can test also with the latest Android versions ?

**Reply by devemux86** [reply]
Some phones can wake up with a double tap on their screen.

**Comment by DolomitesTom** [comment]
Cool functionality

**Comment by devemux86** [comment, marked as answer]
Turn on / off screen in navigation is available in [**Cruiser 5.6.2**](https://github.com/devemux86/cruiser/discussions/834).

See the instructions [above](https://github.com/devemux86/cruiser/discussions/61#discussioncomment-13816480).

**Comment by 0709wiwiwi** [comment]
First post updated.  The posted workaround suggestion to just dim theLCD  backlight to minimal has been removed.
The display OFF/ON feature is less complex and can be done without device admin rights since A9.

_To use the proximity sensor in this way, the necessary hardware must, of course, be present in the phone._
_Some recent Samsung phones lack the sensor hardware required for this purpose._
